ignite/engine/engine.py

```diff
@@ -89,18 +91,19 @@ def __init__(self, process_function):
 
         self._check_signature(process_function, 'process_function', None)
 
-    def register_events(self, *event_names):
+    def register_events(self, *event_names, event_to_attr=None):
         """Add events that can be fired.
 
         Registering an event will let the user fire these events at any point.
         This opens the door to make the :meth:`~ignite.engine.Engine.run` loop even more
         configurable.
 
-        By default, the events from :class:`~ignite.engine.Events` are registerd.
+        By default, the events from :class:`~ignite.engine.Events` are registered.
 
         Args:
             *event_names: An object (ideally a string or int) to define the
                 name of the event being supported.
+            event_to_attr (dict/object): A dictionary or object to map an event to a state attribute.
```

Comment:
> dict/object): A dictionary or object to map an event to a state attribute.

This is not clear what is the type of event_to_attr.

Reply:
Updated, event_to_attr should only be a dictionary, code includes check. 

Python 2.7 does not allow for regular parameter after * parameter, so had to update the code accordingly.